src/util/evalUnits.ts

```diff
@@ -40,11 +44,25 @@ const alphaInts = /^[0-9]*[a-zA-Z]{1}$/;
 
 export enum EvalMessages {
   GIBBERISH = 'Input is not correct. Use numbers, floats or expression (e.g. 1k, 1.3m)',
+  ZERO = 'You cannot send 0 funds',
+  NEGATIVE = 'You cannot send negative amount of funds',
   SUCCESS = '',
   SYMBOL_ERROR = 'Provided symbol is not correct',
   GENERAL_ERROR = 'Check your input. Something went wrong'
 }
 
+const transformShorthandToBN = (input: string): BN | null => {
+  // find the character from the alphanumerics
+  const charPart = input.replace(/[0-9.,]/g, '');
+  // find the value from the si list
+  const siVal = si.find((s) => s.symbol === charPart);
+  // unfortunately in the case of shorthand parseFloat has to be used
+  // as BN library does not support decimals
+  const numbered = parseFloat(input.replace(/[,]/g, '.'));
+  // get only the numeric parts of input
+  return siVal ? new BN(numbered * siVal.value) : null;
```

Comment:
The multiplication here can still overflow the JS type size unfortunately. I think we will need to complicate this a bit more to make it fully fool proof. I'd suggest something like:
1. Detect if there are any `,` and `.`
2. If not - parse using BN instead of `parseFloat` and perform multiplication on BN.
3.  Otherwise:
4. Try to `parseFloat` and compare `parseFloat(input).toString() === input` to avoid trimming any data - if that's not correct then display error about unsupported notation. (Note that the `toString` will turn to scientific notation at some point but IMHO it's fine).

Basically the premise for this is that we either want to parse whatever the user types exactly as intended or display an error. Note that decimals are most likely going to be used for small values. Like I don't expect anyone to write something like `10000000000000000.1 M` or `1.00000000000000000001 E`, but if they do we should make it clear that it's not supported and they need to use non-float notation for this.

Reply:
Issue was actually crazy:
```
  // This may seem crazy but 1e21 and 1e24 were actually passed to BN like that and making it break.
  // (while 1e18 is transforming to format 1000000000)
  ```